frontend/src/components/question-creation/question-elements/modals/short-answer/ShortAnswerModal.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalCloseButton,
+  Text,
+  Divider,
+  ModalBody,
+  FormLabel,
+  FormControl,
+  Input,
+  FormErrorMessage,
+} from "@chakra-ui/react";
+
+interface ShortAnswerModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onConfirm: (data: number) => void;
+  data?: number;
+}
+
+const ShortAnswerModal = ({
+  isOpen,
+  onClose,
+  onConfirm,
+  data,
+}: ShortAnswerModalProps): React.ReactElement => {
+  const [answer, setAnswer] = useState<number>();
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setAnswer(data);
+  }, [data]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const input = parseInt(event.target.value, 10);
+    const updatedAnswer = Number.isNaN(input) ? undefined : input;
+    setAnswer(updatedAnswer);
+  };
+
+  const handleClose = () => {
+    setAnswer(data);
+    setError(false);
+    onClose();
+  };
+
+  const handleConfirm = () => {
+    if (typeof answer !== "undefined") {
+      onConfirm(answer);
+      handleClose();
+    } else {
+      setError(true);
+    }
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={handleClose} isCentered>
+      <ModalOverlay />
+      <ModalContent minWidth="42vw">
+        <ModalHeader>
+          <Text textStyle="subtitle2">Create short answer question</Text>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <FormControl isRequired isInvalid={error}>
+            <FormLabel color="grey.300" style={{ fontSize: "18px" }}>
+              Enter correct answer
+            </FormLabel>
+            <Input
```

Comment:
seems like now we can't take in decimals

Reply:
hmmm but do students really need to learn about decimals? 🤔

Reply:
just kidding ill take a look